sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs

```diff
@@ -18,6 +18,15 @@ public enum ResponseStatusOption
 }
 namespace Azure.Core
 {
+    public sealed partial class AzureError
```

Comment:
Is this concept specific to Azure? i.e. should this be called ServiceError?

Reply:
It was called `ServiceError` in the first iteration. This exact shape is declared Azure REST guidenline as is Azure-specific.

Reply:
I'd prefer not to call it `ServiceError` because that'll cause a lot of confusion for anyone trying to search: https://docs.microsoft.com/en-us/dotnet/api/?term=serviceerror

Reply:
> that'll cause a lot of confusion for anyone trying to search

Good point. Thought, I would still like to avoid using "Azure" in general purpose APIs. Maybe ServiceResponseError? or just ResponseError?

Reply:
>docs.microsoft.com/en-us/dotnet/api/?term=serviceerror

I don't think there are that many search results. For example, https://docs.microsoft.com/en-us/dotnet/api/?term=Response didn't stop us.

Reply:
I kind of like either `RequestError` to tie it to `RequestFailedException` or `ResponseError` to tie it to `Response*`.

Reply:
`RequestError` sounds like it's a validation error (the request is wrong).

`ErrorResponse`?

Reply:
I think ErrorResponse implies that it's a subtype of Response.

Reply:
I like the `ServiceError` the most. It describes exactly what the type is about.